### PR TITLE
Change LHN header to Expensify

### PR DESCRIPTION
**Pull request by bondydaa**
### Details

This updates the "header" to show `Expensify` per the design mocks on the linked issue. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/18382

### Tests

**LHN**
1. Sign in to new dot
2. Confirm the top of the LHN says `Expensify` and not `Chats` and has the proper environment badge

**Signin**
1. ensure you're logged out 
2. load new dot
3. ensure the Expensify wordmark shows in green with the proper environment badge

You can test the various badges locally by editing: https://github.com/Expensify/App/blob/a29631b947530833f4722adf56fa9e267850176b/src/components/ExpensifyWordmark.js#L41 and use the constants from above: `CONST.ENVIRONMENT.DEV`, `CONST.ENVIRONMENT.STAGING`, `CONST.ENVIRONMENT.PRODUCTION`

- [x] Verify that no errors appear in the JS console

### Offline tests

Nothing specific really but same it should say `Expensify` not `Chat` at top of LHN

### QA Steps

**LHN**
1. Sign in to new dot
2. Confirm the top of the LHN says `Expensify` and not `Chats` and has the proper environment badge

**Signin**
1. ensure you're logged out 
2. load new dot
3. ensure the Expensify wordmark shows in green with the proper environment badge

Wash, rinse, repeat for Staging and Production

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

### Dev
**Signin**
image
image

**LHN**
Screenshot 2023-05-08 at 4 08 44 PM
Screenshot 2023-05-08 at 4 08 34 PM

### Staging
**Signin**
image
image

**LHN**
image

image

### Production
**Signin**

**LHN**

</details>

<details>
<summary>Mobile Web - Chrome</summary>

### Dev
**Signin**
image

**LHN**
Screenshot 2023-05-08 at 4 08 20 PM
Screenshot 2023-05-08 at 4 08 27 PM

### Staging
**Signin**
image
image

**LHN**
image
image

### Production
**Signin**

**LHN**

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

### Dev
**Signin**

**LHN**

### Staging
**Signin**

**LHN**

### Production
**Signin**

**LHN**

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Review by parasharrajat — CHANGES_REQUESTED** [review]
We have to use the wordmark SVG https://github.com/Expensify/App/blob/3585329ec6fe8a8bed127fc44b06ff698380a440/assets/images/expensify-wordmark.svg  instead of Text as per https://github.com/Expensify/App/issues/18382#issuecomment-1534812446

**Comment by bondydaa** [comment]
okay if you look at these 2 commits 

https://github.com/Expensify/App/pull/18449/commits/3b461e23899c6ee5b1cefc5caab4d51ab0218f87
https://github.com/Expensify/App/pull/18449/commits/735725c87039a142577550ea64788fdd9c93a586

that's how I first tried to get this to work as per the thread here: https://expensify.slack.com/archives/C01GTK53T8Q/p1683315638909159

But that led to this linter error:
```
/Users/dbondy/Expensidev/App/src/pages/home/sidebar/SidebarLinks.js
  164:25  error  Do not pass children as props. Instead, nest children between the opening and closing tags  react/no-children-prop
```

So that's why I changed it and instead have a new component that is able to just render it's children.

We could also refactor `Header` to render it's children but that would require also updating everywhere `HeaderWithCloseButton` is used and I'm not sure we want to do that right now.

**Comment by bondydaa** [comment]
also I still can't get iOS or android to build locally so I can't test there 😢 I'll see if I can get that figured out next week.

**Comment by shawnborton** [comment]
How about we use the wordmark logos we have that already have the staging/dev/ad hoc labels baked in? Like this from the sign in page:
image

**Comment by bondydaa** [comment]
oh we have a wordmark component already 😅 https://github.com/Expensify/App/blob/0bc7607bf3/src/components/ExpensifyWordmark.js#L32 pretty sure I can just use that then

**Review by parasharrajat — COMMENTED** [review]
Is It ready?

**Comment by bondydaa** [comment]
@parasharrajat yeah this is ready at least to get an initial review please!

@shawnborton could use some CSS help, the production one is slightly off (a small indent to the right), I don't see any `margin` or `padding` being added so I think it might be a flexbox thing but 🤷 
image
image
image

File: src/components/ExpensifyWordmark.js [hunk 1]
```diff
@@ -9,13 +10,23 @@ import withEnvironment, {environmentPropTypes} from './withEnvironment';
...
-import styles from '../styles/styles';
 import * as StyleUtils from '../styles/StyleUtils';
 import variables from '../styles/variables';
 
 const propTypes = {
     ...environmentPropTypes,
     ...windowDimensionsPropTypes,
+
+    /** The styles to apply for the View wrapping the svg */
+    containerStyles: PropTypes.arrayOf(PropTypes.object),
```
Comment (bondydaa) [inline src/components/ExpensifyWordmark.js:21]: This is failing the linter but I see it's how we do this elsewhere in the code base so not sure if I should ignore this like we do here or if there's some other proptypes i should use? https://github.com/Expensify/App/blob/8f1617419ac34ba40bc4e4bfd4ed7fd1de0f12cd/src/components/FormAlertWithSubmitButton.js#L12-L14

**Comment by bondydaa** [comment]
throwing @grgia on here too since she's been pretty involved with the theming stuff and i think initially added the SVGs that I'd just remove a couple styles from.

**Comment by shawnborton** [comment]
> @shawnborton could use some CSS help, the production one is slightly off (a small indent to the right), I don't see any margin or padding being added so I think it might be a flexbox thing but 🤷

Is there any reason why we need to touch those images? I think you should just leave those images as they are, and then for the LHN let's just replace the word "Chats" with a wordmark SVG (which already exists in the repo) and then use CSS to style up the existing environment badges to make them appear to be smaller.

**Comment by bondydaa** [comment]
It seems silly to me have 2 different ways of displaying the Wordmark and environment badge:  One for the sign in pages (which is a SVG with both wordmark and environment badge) and then for LHN of SVG + Environment badge component. 

The way I have it avoids needing to throw a lot of custom logic into the Header component just for the LHN or avoid having 2 components that are basically the same with slightly different usages. 

Are the SVG images I used the wrong font-family for the wordmark?

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by bondydaa** [comment]
Alrighty fixed the proptypes, only thing is the production version of the wordmark being slightly indented... maybe it has to do with the width of the element & flexbox since when the badges are present it's aligned properly?

**Comment by shawnborton** [comment]
Nice, that looks better. The idea was that the wordmark portion would be 108px wide, and then we have to account for the extra width for the versions that have badges.

**Comment by shawnborton** [comment]
> Alrighty fixed the proptypes, only thing is the production version of the wordmark being slightly indented...

Is this still happening? And can you confirm that the existing sign up page looks exactly like it does on production too?

**Comment by bondydaa** [comment]
Hmm just had a thought... @parasharrajat do you want to take over this PR and get it across the finish line? There no reason to have me a be a blocker here while I'm focused on Manual Request stuff. 

if so feel free to assign yourself and move me to a reviewer.

**Comment by bondydaa** [comment]
oh hmm... wonder if we could CP the changes over or something if you give me write access to your repo/fork/branch? whatever is easiest just let me know

**Comment by bondydaa** [comment]
or actually I can also just post the diff and you can `git apply` it too. 1 sec

**Comment by bondydaa** [comment]
hmm weird what? 

I merged https://github.com/Expensify/App/pull/18802 not this

**Comment by bondydaa** [comment]
oh interesting! whoops yes I def should have closed this first 🤦

**Pull request merged**